sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,645 @@
+ *         puts("Error creating session");
+ *         sock_dtls_close(&dtls_sock);
+ *         sock_udp_close(&udp_sock);
+ *         return -1;
+ *     }
+ *
+ *     const char data[] = "HELLO";
+ *     int res = sock_dtls_send(&dtls_sock, &session, data, sizeof(data));
+ *     if (res >= 0) {
+ *         printf("Sent %d bytes: %*.s\n", res, res, data);
+ *         res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT);
+ *         if (res > 0) {
+ *             printf("Received %d bytes: %*.s\n", res, res, rcv);
+ *         }
+ *     }
+ *     else {
+ *         puts("Error sending data");
+ *     }
+ *
+ *     sock_dtls_session_destroy(&dtls_sock, &session);
+ *     sock_dtls_close(&dtls_sock);
+ *     sock_udp_close(&udp_sock);
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * This is an example of a DTLS echo client.
+ *
+ * Like the server, we must first create the UDP sock.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_t udp_sock;
+ * sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ * local.port = 12345;
+ * sock_udp_create(&udp_sock, &local, NULL, 0);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After that, we set the address of the remote endpoint where the packet is
+ * supposed to be sent to and the port the server is listening to which is
+ * DTLS_DEFAULT_PORT (20220).
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_ep_t remote;
+ * remote.port = DTLS_DEFAULT_PORT;
+ * remote.netif = gnrc_netif_iter(NULL)->pid;   // only if GNRC_NETIF_NUMOF == 1
+ *
+ * if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, SERVER_ADDR)) {
+ *     puts("Error parsing destination address");
+ *     return -1;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After the UDP sock is created, we can proceed with creating the DTLS sock.
+ * Before sending the packet, we must first create a session with the remote
+ * endpoint using @ref sock_dtls_session_create(). If the handshake is
+ * successful and the session is created, we send packets to it using
+ * @ref sock_dtls_send(). If the packet is successfully sent, we listen for
+ * the response with @ref sock_dtls_recv().
+ *
+ * As @ref sock_dtls_create() and @ref sock_dtls_close() only manages the DTLS
+ * layer, we still have to clean up the created UDP sock from before by calling
+ * @ref sock_udp_close() on our UDP sock in case of error or we reached the end
+ * of the application.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * char rcv[128];
+ * sock_dtls_t dtls_sock;
+ * sock_dtls_session_t session;
+ *
+ * [...]
+ *
+ * if (sock_dtls_create(&dtls_sock, &udp_sock,
+ *                      SOCK_DTLS_CLIENT_TAG,
+ *                      SOCK_DTLS_1_2, SOCK_DTLS_CLIENT) < 0) {
+ *     puts("Error creating DTLS sock");
+ *     sock_udp_close(&udp_sock);
+ *     return -1;
+ * }
+ *
+ * if (sock_dtls_session_create(&dtls_sock, &remote, &session) < 0) {
+ *     printf("Error creating session: %d\n", (int)res);
```

Comment:
`res` is not assigned with the result of `sock_dtls_session_create()`.

Reply:
I missed that. Removed `res`.